cvat-core/src/frames.ts

```diff
@@ -3,28 +3,35 @@
 //
 // SPDX-License-Identifier: MIT
 
-import { isBrowser, isNode } from 'browser-or-node';
-
-import * as cvatData from 'cvat-data';
-import { DimensionType } from 'enums';
+import _ from 'lodash';
+import {
+    FrameDecoder, BlockType, DimensionType, decodeContextImages, RequestOutdatedError,
+} from 'cvat-data';
 import PluginRegistry from './plugins';
 import serverProxy, { RawFramesMetaData } from './server-proxy';
-import {
-    Exception, ArgumentError, DataError, ServerError,
-} from './exceptions';
+import { Exception, ArgumentError, DataError } from './exceptions';
 
 // frame storage by job id
 const frameDataCache: Record<string, {
-    meta: RawFramesMetaData;
+    meta: RawFramesMetaData & { deleted_frames: Record<number, boolean> };
     chunkSize: number;
     mode: 'annotation' | 'interpolation';
     startFrame: number;
     stopFrame: number;
-    provider: cvatData.FrameProvider;
-    frameBuffer: FrameBuffer;
+    decodeForward: boolean;
+    forwardStep: number;
+    latestFrameDecodeRequest: number | null;
+    latestContextImagesRequest: number | null;
+    provider: FrameDecoder;
     decodedBlocksCacheSize: number;
-    activeChunkRequest: null;
-    nextChunkRequest: null;
+    activeChunkRequest: Promise<void> | null;
+    activeContextRequest: Promise<Record<number, ImageBitmap>> | null;
+    contextCache: Record<number, {
+        data: Record<number, ImageBitmap>;
+        timestamp: number;
+        size: number;
+    }>;
+    getChunk: (chunkNumber: number, quality: 'original' | 'compressed') => Promise<ArrayBuffer>;
```

Comment:
Maybe create enum for `quality`, Ive seen `quality: 'original' | 'compressed'` in several places

Reply:
Added